README.md

```diff
@@ -2,4 +2,329 @@
+
+## `Array`
+
+### Type Annotation
+
+To explicitly declare the type of an empty array, wrap the type within square brackets `[` `]`. You can use either type annotation to set it equal to empty square brackets, or the initializer syntax that uses a parenthesis following the array type:
+
+```swift
+let array: [Element] = []      // type annotation
+let array: array<Element> = [] // long form annotation
+let array = [Element]()        // initializer
+```
+
+To explicitly declare an array with contents, the contents must match the annotated type.
+
+```swift
+var instructors: [String] = ["Joe", "Tim", "Jim", "Tom"]
+let primes: [Int] = [1, 2, 3, 5, 7, 11]
+let sqrts: [Double] = [1, 1.414, 1.732, 2.236, 2.646, 3.317]
+```
+
+### Subscripting
+
+Arrays can be subscripted by passing an index integer into a pair of square brackets suffixed to the array's name:
+
+```swift
+// getting
+
+let joe = instructors[0]
+print(joe)
+print(instructors[1])
+```
+This will print:
+
+```
+Joe
+Tim
+```
+
+```swift
+// setting, only for arrays declared with var
+
+instructors[2] = "Jim Campagno"
+primes[5] = 13   // error
+```
+### Methods
+
+#### `.append()`
+
+You can add to mutable array using the `.append()` method with the element to add placed inside the parenthesis. The element must match the array's type.
+
+```swift
+instructors.append("Mark")
+instructors.append(42)   // error
+primes.append(13)  // error
+```
+
+#### `.removeLast()`
+
+The `.removeLast()` method will return the element at the highest index of array, however it will cause a crash if the array is empty. You must protect against calling this method on array when it is empty when you implement this method, otherwise use `.popLast()` which returns an optional.
+
+#### `.popLast()`
+
+The `.popLast()` method returns an optional of the array's content type, which will need to be unwrapped. This method will not cause a run time crash if it is called on an empty array since it will return an optional containing `nil` in such a case.
+
+### `.reverse()`
+
+The `.reverse()` method returns a copy of the array with the elements in reverse order. It does **not** implicitly mutate the array it is called on.
+
+### Array Properties
+
+#### `.first` & `.last`
+
+The `.first` and `.last` are properties which return on optional containing the element at the array's lowest index (`.first`) or at its highest index (`.last`). If the array is empty, accessing these properties will return an optional containing `nil`.
+
+#### `.count`
+
+Arrays have a `count` property which returns an `Int` count of the number of elements currently in its membership.
+
+### Arrays with Loops
+
+To iterate over an array without needing the index number, use a `for` loop:
+
+```swift
+for Element in Array {
+    // statements
+}
+```
+
+An implementation of this might look like:
+
+```swift
+let instructors: [String] = ["Joe", "Tim", "Jim", "Tom", "Mark"]
```

Comment:
In general we want to encourage people to use inferred types. So in this case we don't need the `[String]` becuase it's obvious that this would create an array of `String` objects